validatorobserver/validator_observer.go

```diff
@@ -0,0 +1,197 @@
+package validatorobserver
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/Ethernal-Tech/apex-bridge/common"
+	"github.com/Ethernal-Tech/apex-bridge/contractbinding"
+	"github.com/Ethernal-Tech/apex-bridge/eth"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/hashicorp/go-hclog"
+)
+
+type ValidatorSetObserver struct {
+	validatorSetStream  chan *Validators
+	validators          Validators
+	validatorSetPending bool
+	bridgeSmartContract eth.IBridgeSmartContract
+	logger              hclog.Logger
+}
+
+type ValidatorsChainData struct {
+	Keys       []eth.ValidatorChainData
+	SlotNumber uint64
+}
+
+type Validators struct {
+	Data map[string]ValidatorsChainData
+}
+
+const (
+	timeout = 1 * time.Second
+)
+
+var lock = sync.RWMutex{}
+
+func NewValidatorSetObserver(
```

Comment:
Registered chains can be changed overtime. Can we make at least this intialization as a separate function? I think we should at least sometimes call this to check if there are new registered chains. Probably as a part of execute

Reply:
But if it too heavy for each call, at least make room to call sometimes after initialization